File: client/travelagency/src/app/shared/models/agency.ts
import { PlanListModel } from "./plan";

export interface AgencyListModel {
    id: number;
    name: string;
    address: string;
    phoneNumber: string;
    email?: string;
    accountNumber?: string;
}

export interface AgencyCreateModel {
    name: string;
    address: string;
    phoneNumber: string;
    email?: string;
    accountNumber?: string;
}

export interface AgencyDetailsModel {
    id: number;
    name: string;
    address: string;
    phoneNumber: string;
    email: string | null;
    accountNumber: string | null;
    plans: PlanListModel[];
}

File: server/TravelAgency.Domain/Models/Agency.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TravelAgency.Domain.Models
{
    public class Agency : BaseEntity
    {
        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;

        [Required]
        [MaxLength(100)]
        public string Address { get; set; } = string.Empty;

        [Required]
        [MaxLength(50)]
        public string PhoneNumber { get; set; } = string.Empty;

        [MaxLength(50)]
        public string? Email { get; set; }

        [MaxLength(100)]
        public string? AccountNumber { get; set; }

        [Required]
        public int OrganizationId { get; set; }

        [ForeignKey(nameof(OrganizationId))]
        public Organization Organization { get; set; }

        [InverseProperty("Agency")]
        public List<Plan> Plans { get; set; } = new();
    }
}
